### ISSUE TITLE
Feature idea: better support for interlaced videos (at least field-based, if not frame-based)

### ISSUE THREAD

**Issue by JohnstonJ** [Asd-g/AviSynth-FillBorders#5]
Currently, the plugin fails with an error if an interlaced video is used as input to FillBorders, even if the fields were already separated: https://github.com/Asd-g/AviSynth-FillBorders/blob/34cb4a7421ff2d1d7b5294543648b088e973c908/src/FillBorders.cpp#L649-L652

Obviously, a naive attempt at filling borders in a frame-based interlaced video will not work correctly if special handling for interlaced frames is not provided.

However, I think it's not hard to make a reasonable attempt at handling interlaced videos.  This prototype code provides a reasonable demonstration of the concept:

```
function DoFillBorders( \
    clip c, int "left", int "top", int "right", int "bottom", int "mode", bool "interlaced" \
) {
    Assert(top % 2 == 0 && bottom % 2 == 0, "FillBorders requires even top/bottom.")
    last = c
    if (interlaced) {
        # FillBorders will refuse to work, even after calling SeparateFields(), unless
        # we first mark the individual frame properties as progressive.
        SeparateFields().propSetInt("OldFieldBased", function[](clip c) {
            return c.propGetInt("_FieldBased")
        })
        propSetInt("_FieldBased", function[](clip c) { return 0 })
        top = top / 2
        bottom = bottom / 2
    }
    FillBorders(left=left, top=top, right=right, bottom=bottom, mode=mode)
    if (interlaced) {
        propSetInt("_FieldBased", function[](clip c) {
            return c.propGetInt("OldFieldBased")
        }).propDelete("OldFieldBased").Weave()
    }
}

DoFillBorders(5, 10, 15, 20, 6, true)
```

In short:

1.  Call `SeparateFields`
2.  Run the normal FillBorders code, but with the fill margins cut in half for top and bottom.
3.  Call `Weave` to put it back together.

Unfortunately, it seems that there's a usability issue here: if I don't explicitly deal with the `_FieldBased` frame property after calling `SeparateFields`, then the function refuses to process the clip, even though the fields have been separated and can now be reasonably filled using the normal methods.  So, then I have to write this clunky property setting/clearing code.

**First idea for improvement**

Don't block processing of field-based interlaced videos.  For example, I should be able to run:

```
SeparateFields().FillBorders(left=left, top=top / 2, right=right, bottom=bottom / 2, mode=mode).Weave()
```

without any errors.

**Second idea for improvement**

For even better ease of use, add an `interlaced` parameter to `FillBorders`, and/or automatically handle interlacing when the input is interlaced.  For example:

```
FillBorders(left=left, top=top, right=right, bottom=bottom, mode=mode, interlaced=true)
```

**Comment by Asd-g** [comment]
What mode you're using for interlaced content?

**Comment by JohnstonJ** [comment]
@Asd-g The above example uses mode 6 with FillBorders, but I think any mode can reasonably work when the fields are separated.  Here's an example input video information: 
![image](https://github.com/user-attachments/assets/c82b8424-823f-432c-855d-b814ea112ef8)

And frame properties:
![image](https://github.com/user-attachments/assets/9b18ff44-9a20-4e4f-b4e5-8cf8741ab2df)

This is after calling `AssumeBFF`.

Notably, it's frame-based (interleaved).  I have found that the `DoFillBorders` function above that I posted earlier seems to provide reasonable results - i.e. use SeparateFields (to convert from frame-based to field-based), FillBorders, then Weave.  But it's cumbersome - especially the part where I have to manually mess with `_FieldBased` frame properties.

There is another filter, SpotLess, that I have been using which also does not directly support interlaced input.  But with that one, simply separating fields seems to be enough for the filter to run (no need to mess with frame properties).
